api/renter.go

```diff
@@ -83,9 +99,32 @@ type (
 
 // renterHandlerGET handles the API call to /renter.
 func (api *API) renterHandlerGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	// calculate financial metrics from contracts
+	periodStart := api.renter.CurrentPeriod()
+	var fm RenterFinancialMetrics
+	for _, c := range api.renter.Contracts() {
+		if c.StartHeight < periodStart {
+			continue
+		}
+		fm.ContractSpending = fm.ContractSpending.Add(c.TotalCost)
+		fm.DownloadSpending = fm.DownloadSpending.Add(c.DownloadSpending)
+		fm.UploadSpending = fm.UploadSpending.Add(c.UploadSpending)
+		fm.StorageSpending = fm.StorageSpending.Add(c.StorageSpending)
+		fm.Unspent = fm.Unspent.Add(c.RenterFunds())
+	}
+	settings := api.renter.Settings()
+	fm.Unspent = fm.Unspent.Add(settings.Allowance.Funds)
+	if fm.Unspent.Cmp(fm.ContractSpending) < 0 {
+		// avoid negative currency error
+		fm.Unspent = types.ZeroCurrency
+	} else {
+		fm.Unspent = fm.Unspent.Sub(fm.ContractSpending)
+	}
```

Comment:
Might be more obvious to write it like this:
```go
fm.Unspent = settings.Allowance.Funds
for _, c := range contracts {
    // subtract the total amount we spent on the contract, but add back
    // the amount in the contract that is still spendable
    fm.Unspent = fm.Unspent.Sub(c.TotalCost).Add(c.RenterFunds())
}
```
(of course, you still need a negative check for the `Sub`)